src/streamlink/plugins/afreeca.py

```diff
@@ -225,9 +228,12 @@ def _get_streams(self):
         self.title = channel.get("TITLE")
 
         for qkey in self.QUALITYS:
-            hls_stream = self._get_hls_stream(broadcast, username, qkey, rmd)
+            hls_stream = self._get_hls_stream(broadcast, username, qkey, rmd, stream_password)
             if hls_stream:
                 yield qkey, hls_stream
+            elif channel.get("BPWD") == "Y":
+                log.error("Stream is Password-Protected")
+                return
```

Comment:
It'd be a bit cleaner to collect all streams in a `dict[str, HLSStream]` or `list[tuple[str, HLSStream]]` and return that instead of turning `_get_streams()` into a generator. Then you can check this data collection after the stream quality iteration and log an error if no streams are available.

Reply:
```suggestion
        if not streams and channel.get("BPWD") == "Y":
            log.error("Stream is Password-Protected")
            return
```

I meant that the pw-protection log message should be moved out of the loop, because you're returning `None` on the very first missing stream result.